File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<int[]> queue = new LinkedList<>();
        ArrayList<Integer> deployedPerDay = new ArrayList<>();
        
        for(int i = 0; i < progresses.length; i++){
            queue.add(new int[] {progresses[i], i});
        }
        
        int curDeployIdx = 0;
        
        while(!queue.isEmpty()){
            int queueSize = queue.size();
            int deployed = 0;

            for(int i = 0; i < queueSize; i++){
                int[] curInfo = queue.poll();
                curInfo[0] += speeds[curInfo[1]];
                
                if (curInfo[0] >= 100 && curDeployIdx == curInfo[1]){
                    curDeployIdx++;
                    deployed++;
                } else {
                    queue.add(curInfo);
                }
            }
            if(deployed > 0) deployedPerDay.add(deployed);
        }
            
        int[] answer = new int[deployedPerDay.size()];
        
        for(int i = 0; i < answer.length; i++){
            answer[i] = deployedPerDay.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    n = len(progresses)
    left_day = [math.ceil((100 - progresses[i]) / speeds[i]) for i in range(n)]  # 올림 연산
    answer = []
    
    longest_day = left_day[0]
    count = 0
    
    for i in range(n):        
        if longest_day >= left_day[i]:
            count += 1
        else:
            answer.append(count)
            count = 1
            longest_day = left_day[i] 
    
    answer.append(count)    
    return answer